tests/subsys/dfu/dfu_target_stream/src/main.c

```diff
@@ -100,7 +95,7 @@ static void test_dfu_target_stream(void)
 }
 
 #ifdef CONFIG_DFU_TARGET_STREAM_SAVE_PROGRESS
-static void test_dfu_target_stream_save_progress(void)
+ZTEST(dfu_target_stream_test, test_dfu_target_stream_save_progress)
 {
```

Comment:
You can consider using in this test `Z_TEST_SKIP_IFNDEF` macro:
https://github.com/zephyrproject-rtos/zephyr/blob/bb6fa5707f77d72307008405f19dd83247405392/subsys/testsuite/ztest/include/zephyr/ztest_test_new.h#L370-L379
like:
```c
Z_TEST_SKIP_IFNDEF(CONFIG_DFU_TARGET_STREAM_SAVE_PROGRESS);
```
Thanks to this you could avoid using `#ifdef` directive and separate test definition dedicated for test skipping.